homeassistant/components/http/security_filter.py

```diff
@@ -57,35 +57,36 @@ async def security_filter_middleware(
         request: Request, handler: Callable[[Request], Awaitable[StreamResponse]]
     ) -> StreamResponse:
         """Process request and block commonly known exploit attempts."""
-        path_with_query_string = f"{request.path}?{request.query_string}"
+        rel_url = request.rel_url
+        path_with_query_string = _recursive_unquote(rel_url.raw_path_qs)
```

Comment:
No point in fetching the unquoted `.path` and `.query_string` if we are always going to unquote it anyways